apps/web/vibes/soul/form/file-input/file-dropzone.tsx

```diff
@@ -0,0 +1,323 @@
+  const onDrop = useCallback(
+    (event: DragEvent<HTMLDivElement>) => {
+      onDropProp?.(event);
+
+      if (event.defaultPrevented) return;
+
+      event.preventDefault();
+      setIsDragging(false);
+      updateInputFiles(Array.from(event.dataTransfer.files));
+    },
+    [onDropProp, updateInputFiles],
+  );
+
+  const onDragEnter = useCallback(
+    (event: DragEvent<HTMLDivElement>) => {
+      onDragEnterProp?.(event);
+
+      if (event.defaultPrevented) return;
+
+      event.preventDefault();
+      setIsDragging(true);
+    },
+    [onDragEnterProp],
+  );
+
+  const onDragLeave = useCallback(
+    (event: DragEvent<HTMLDivElement>) => {
+      onDragLeaveProp?.(event);
+
+      if (event.defaultPrevented) return;
+
+      const relatedTarget = event.relatedTarget;
+      if (
+        relatedTarget &&
+        relatedTarget instanceof Node &&
+        event.currentTarget.contains(relatedTarget)
+      ) {
+        return;
+      }
+
+      event.preventDefault();
+      setIsDragging(false);
+    },
+    [onDragLeaveProp],
+  );
+
+  const onPaste = useCallback(
+    (event: ClipboardEvent<HTMLDivElement>) => {
+      onPasteProp?.(event);
+
+      if (event.defaultPrevented) return;
+
+      event.preventDefault();
+      setIsDragging(false);
+
+      const items = event.clipboardData.items;
+      if (items.length === 0) return;
+
+      const pastedFiles: File[] = [];
+
+      for (const item of items) {
+        if (item.kind === 'file') {
+          const file = item.getAsFile();
+          if (file) {
+            pastedFiles.push(file);
+          }
+        }
+      }
+
+      if (pastedFiles.length === 0) return;
+
+      const inputElement = inputRef.current;
+      if (!inputElement) return;
+
+      const dataTransfer = new DataTransfer();
+      for (const file of pastedFiles) {
+        dataTransfer.items.add(file);
+      }
+
+      inputElement.files = dataTransfer.files;
+      inputElement.dispatchEvent(new Event('change', { bubbles: true }));
+    },
+    [inputRef, onPasteProp],
+  );
+
+  const onKeyDown = useCallback(
+    (event: KeyboardEvent<HTMLDivElement>) => {
+      onKeyDownProp?.(event);
+
+      if (!event.defaultPrevented && (event.key === 'Enter' || event.key === ' ')) {
+        event.preventDefault();
+        inputRef.current?.click();
+      }
+    },
+    [inputRef, onKeyDownProp],
+  );
+
+  const onClick = useCallback(
+    (event: MouseEvent<HTMLButtonElement | HTMLDivElement>) => {
+      if (event.defaultPrevented) return;
+
+      inputRef.current?.click();
+    },
+    [inputRef],
+  );
+
+  const onChange = useCallback(
+    (event: ChangeEvent<HTMLInputElement>) => {
+      const files = Array.from(event.target.files ?? []);
+      onFilesAdded(files);
+
+      if (inputRef.current) {
+        inputRef.current.value = '';
+      }
+    },
+    [onFilesAdded],
+  );
+
+  return (
+    <>
+      <Label className={clsx(hideLabel === true ? 'sr-only' : 'mb-2')} htmlFor={labelId}>
```

Comment:
super nit: generally when a value is explicitly a boolean, even if it's optional, we don't compare it to an explicit value.